imv: install .desktop file

wizeman opened the pull request:
###### Motivation for this change

Install upstream's `imv.desktop` file.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

wizeman committed:
imv: install .desktop file

rnhmjoj commented on pkgs/applications/graphics/imv/default.nix:
```diff
@@ -90,6 +90,11 @@ stdenv.mkDerivation rec {
     ++ builtins.map (b: backends."${b}") withBackends;
 
+  postInstall = ''
+    mkdir -p $out/share/applications
+    cp ../files/imv.desktop $out/share/applications/
```
```suggestion
    # fix the executable path and install the desktop item
    substituteInPlace ../files/imv.desktop --replace "imv %F" "$out/bin/imv %F"
    install -Dm644 ../files/imv.desktop $out/share/applications/
```
It's better to use a store path to make sure the result is pure.

rnhmjoj commented:
Can you please squash the two commits?

r-rmcgibbo commented:
Result of `nixpkgs-review pr 126762` at 34e46d15 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built successfully:</summary>
  <ul>
    <li>imv</li>
  </ul>
</details>
<details>
  <summary>2 suggestions:</summary>
  <ul>
    <li>warning: unused-argument

Unused argument: `fontconfig`.
Near pkgs/applications/graphics/imv/default.nix:8:3:
```
  |
8 | , fontconfig
  |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/applications/graphics/imv/default.nix:113:5:
```
    |
113 |     license = licenses.gpl2;
    |     ^
```
</li>
  </ul>
</details>

wizeman committed:
imv: remove unused fontconfig argument

wizeman committed:
imv: clarify license

licenses.gpl2 is deprecated, either gpl2Only or gpl2Plus should be
specified. However, imv is dual-licensed under the GPL and MIT licenses.

MIT is the one in the root LICENSE file, it's the one that Github shows
in  the main repo page and is also the more permissive one, therefore I
changed it to licenses.mit.

wizeman commented:
I've squashed the commits but I also added a couple of new commits to fix the bot's suggestions regarding the unused argument and the license clarification, please take a look.

Thanks!

rnhmjoj commented:
Changes looks good, thank you.
